synapse/handlers/devicemessage.py

```diff
@@ -100,15 +104,24 @@ def send_device_message(self, sender_user_id, message_type, messages):
 
         message_id = random_string(16)
 
+        context = {"opentracing": {}}
+        opentracing.inject_active_span_text_map(context["opentracing"])
+
         remote_edu_contents = {}
         for destination, messages in remote_messages.items():
-            remote_edu_contents[destination] = {
-                "messages": messages,
-                "sender": sender_user_id,
-                "type": message_type,
-                "message_id": message_id,
-            }
+            with opentracing.start_active_span("to_device_for_user"):
+                opentracing.set_tag("destination", destination)
+                remote_edu_contents[destination] = {
+                    "messages": messages,
+                    "sender": sender_user_id,
+                    "type": message_type,
+                    "message_id": message_id,
+                    "context": json.dumps(context)
```

Comment:
presumably we don't need to do this for each message?

Reply:
We do, there's no guarantee that all the messages from the same context will be sent out at the same time so there would be no way to link the message received back to the original context.